### PR TITLE
Use signed data in stargate client signAndBroadcast

### LINKED ISSUE #561: Need to handle the new sign doc of the `OfflineDirectSigner`
https://github.com/chainapsis/keplr-extension/issues/43
https://github.com/chainapsis/keplr-example/blob/stargate/src/main.js

I am implementing the protobuf tx support for Keplr extension.
For now, signing generally does well.
However, because the current @cosmjs/stargate ignores the `SignDoc` that is given back by the `OfflineSigner`, it is impossible to change the tx config such as the fee, gas from the wallet side.

https://github.com/cosmos/cosmjs/blob/master/packages/stargate/src/signingstargateclient.ts#L126

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #561

### CODE REVIEW

File: packages/stargate/src/encoding.ts [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
   }
   return type;
 }
+
+export function getMsgTypeUrl(type: string): string {
+  const typeUrlEntry = Object.entries(typeRegister).find(([_, value]) => value === type);
+  if (!typeUrlEntry) {
+    throw new Error("Type not known");
+  }
+  return typeUrlEntry[0];
```
Comment (webmaster128) [inline packages/stargate/src/encoding.ts:35]: Just an idea to help explaining what is going on here:

```suggestion
  const [typeUrl, _type] = typeUrlEntry;
  return typeUrl;
```

**Review by webmaster128 — COMMENTED** [review]
Any chance we can test this new functionality? Do we have a signer at hand that can change the signed content?

**Comment by willclarktech** [comment]
> Any chance we can test this new functionality? Do we have a signer at hand that can change the signed content?

@webmaster128 All of our signers just pass the sign doc straight through to the return value. I could add a class that wraps a signer but modifies the fees?

**Comment by webmaster128** [comment]
> @webmaster128 All of our signers just pass the sign doc straight through to the return value. I could add a class that wraps a signer but modifies the fees?

Yeah sounds good. Something that lives in tests only.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +11,8 @@
 - @cosmjs/launchpad: `Secp256k1HdWallet.sign` method renamed `signAmino`.
 - @cosmjs/launchpad-ledger: `LedgerSigner.sign` method renamed `signAmino`.
 - @cosmjs/proto-signing: Add new package for handling transaction signing with
-  protobuf encoding.
+- @cosmjs/proto-signing: Expose `DirectSignResponse` interface. protobuf
+  encoding.
```
Comment (webmaster128) [inline CHANGELOG.md:15]: This looks broken

**Review by webmaster128 — APPROVED** [review]

**Pull request merged**